object-utils.go

```diff
@@ -73,27 +73,32 @@ func IsValidBucketName(bucket string) bool {
 // - Caret ("^")
 // - Grave accent / back tick ("`")
 // - Vertical bar / pipe ("|")
+//
```

Comment:
> no object-name is "stricter" means more restrictions than prefix ok, look at it this way - Valid object names are a subset of valid prefix names.

Okay agreed can you add 'strings.HasPrefix(object, slashSeparator)' as well we should reject such object names as well.